src/envoy/mixer/http_filter.cc

```diff
@@ -0,0 +1,266 @@
+/* Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "precompiled/precompiled.h"
+
+#include "common/common/logger.h"
+#include "common/http/headers.h"
+#include "common/http/utility.h"
+#include "envoy/server/instance.h"
+#include "server/config/network/http_connection_manager.h"
+#include "src/envoy/mixer/http_control.h"
+
+using ::google::protobuf::util::Status;
+using StatusCode = ::google::protobuf::util::error::Code;
+using ::istio::mixer_client::DoneFunc;
+
+namespace Http {
+namespace Mixer {
+namespace {
+
+// Define lower case string for X-Forwarded-Host.
+const LowerCaseString kHeaderNameXFH("x-forwarded-host", false);
+
+// Convert Status::code to HTTP code
+int HttpCode(int code) {
+  // Map Canonical codes to HTTP status codes. This is based on the mapping
+  // defined by the protobuf http error space.
+  switch (code) {
+    case StatusCode::OK:
+      return 200;
+    case StatusCode::CANCELLED:
+      return 499;
+    case StatusCode::UNKNOWN:
+      return 500;
+    case StatusCode::INVALID_ARGUMENT:
+      return 400;
+    case StatusCode::DEADLINE_EXCEEDED:
+      return 504;
+    case StatusCode::NOT_FOUND:
+      return 404;
+    case StatusCode::ALREADY_EXISTS:
+      return 409;
+    case StatusCode::PERMISSION_DENIED:
+      return 403;
+    case StatusCode::RESOURCE_EXHAUSTED:
+      return 429;
+    case StatusCode::FAILED_PRECONDITION:
+      return 400;
+    case StatusCode::ABORTED:
+      return 409;
+    case StatusCode::OUT_OF_RANGE:
+      return 400;
+    case StatusCode::UNIMPLEMENTED:
+      return 501;
+    case StatusCode::INTERNAL:
+      return 500;
+    case StatusCode::UNAVAILABLE:
+      return 503;
+    case StatusCode::DATA_LOSS:
+      return 500;
+    case StatusCode::UNAUTHENTICATED:
+      return 401;
+    default:
+      return 500;
+  }
+}
+
+}  // namespace
+
+class Config : public Logger::Loggable<Logger::Id::http> {
+ private:
+  std::shared_ptr<HttpControl> http_control_;
+  Upstream::ClusterManager& cm_;
+
+ public:
+  Config(const Json::Object& config, Server::Instance& server)
+      : cm_(server.clusterManager()) {
+    std::string mixer_server;
+    if (config.hasObject("mixer_server")) {
+      mixer_server = config.getString("mixer_server");
```

Comment:
you mean to add mixer_server in the cluster config, and in the code to extract IP from cluster_manager to create grpc channel?